src/model.ts

```diff
@@ -186,6 +186,8 @@ class FileBrowserModel implements IDisposable {
    * Refresh the current directory.
    */
   refresh(): Promise<void> {
+    // Refresh the list of kernelspecs and our directory listing.
+    this._getKernelSpecs();
```

Comment:
This might be a hasty add. The `_getKernelSpecs()` method just pushes the results into an internal array. This means repeated calls to it will grow that internal array without bound.